### PR TITLE
Implement client-side auth

### LINKED ISSUE #48: Client-side auth
Enable Auth0 login for the front-end. 

See previous project for guidance:

https://github.com/distributeaid/toolbox/blob/saga/react_ui/src/apollo/client.ts
https://github.com/distributeaid/toolbox/blob/saga/react_ui/src/components/layout/NavBar.tsx#L6-L16
https://github.com/distributeaid/toolbox/blob/saga/react_ui/src/index.tsx#L12-L15

Done when: user can login to auth0 and send JWT-authenticated requests to the back end

### PR DESCRIPTION
**Pull request by deammer**
### Changes
- added `@auth0/auth0-react`
- added the ability to log in and log out using Auth0
- added a public home page
- added a `DropdownMenu` component
- added a bearer token to API calls

### Pixels

|Public home page on mobile|Menu at the top|
|---|---|
|Screen Shot 2021-03-01 at 10 01 43 PM|Screen Shot 2021-03-02 at 9 39 57 PM|

Resolves #48

### CODE REVIEW

File: frontend/src/AppRoot.tsx [hunk 1]
```diff
@@ -1,30 +1,39 @@
...
+import PublicHomePage from './pages/PublicHome'
 import GroupList from './pages/groups/GroupList'
+import PrivateRoute from './components/PrivateRoute'
+
+const AppRoot = () => {
+  const { isAuthenticated, isLoading } = useAuth0()
+
+  if (isLoading) {
+    // TODO make this look a LOT better...
+    return <div>Loading...</div>
```
Comment (deammer) [inline frontend/src/AppRoot.tsx:17]: https://github.com/distributeaid/shipment-tracker/issues/52

File: frontend/src/fixtures/groups.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ const GROUPS: GroupPlaceholder[] = [
   {
     id: 7,
     name: 'Distribute Aid - Leeds',
-    groupType: GroupType.DAHub,
+    groupType: GroupType.DaHub,
```
Comment (deammer) [inline frontend/src/fixtures/groups.ts:72]: Cleaning up my mess from a previous PR 🙈

File: frontend/src/data/apolloClientInstance.ts [hunk 3]
```diff
@@ -1,8 +1,48 @@
...
+const clientUrl = process.env.REACT_APP_CLIENT_URL
+
+// https://www.apollographql.com/docs/react/networking/authentication/
+const authLink = new ApolloLink((operation, forward) => {
+  const { getAccessTokenSilently } = useAuth0()
+
+  return fromPromise(
+    getAccessTokenSilently()
+      .then((token: string) => {
+        console.log(`Access token: ${token}`)
```
Comment (bion) [inline frontend/src/data/apolloClientInstance.ts]: nit: remove console log

File: frontend/src/components/DropdownMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      </div>
+    </div>
+  )
+}
+
+DropdownMenu.Text = DropdownMenuText
+DropdownMenu.Button = DropdownMenuButton
+DropdownMenu.Divider = DropdownMenuDivider
+
+export default DropdownMenu
```
Comment (bion) [inline frontend/src/components/DropdownMenu.tsx:133]: This file is really cleanly laid out 👍🏻

**Review by bion — APPROVED** [review]

File: frontend/src/data/apolloClientInstance.ts (hunk 3 above)
Comment (deammer) [reply frontend/src/data/apolloClientInstance.ts]: Done, thank you!

**Pull request merged**